services/agents/agents.go

```diff
@@ -40,3 +43,28 @@ func redactWords(agent *models.Agent) []string {
 	}
 	return words
 }
+
+func filterOutCollectors(prefix string, args, disabledCollectors []string) []string {
+	isDisabled := func(arg string) bool {
```

Comment:
maybe it's better to create `map[string]struct{}` at the beginning and just check if key is exists instead of looping n*m times.

Reply:
Yes, think on this as well - but without more complex refactoring this does not much bring much effort.
And as addition, we cannot match key by it part (`strings.HasPrefix`) - so it still requires iterations.

Reply:
as far as I see all collectors has only two options:
boolean flag like `--collect.binlog_size`
value flags like `--collect.custom_query.hr.directory=/usr/local/percona/pmm2/collectors/custom-queries/mysql/high-resolution`
for boolean flags we can just concatinate prefix with disabled collector and just compare two values
for value flags we can extract first part before `=` symbol and use it to compare. 
what do you think?

Reply:
Yes, true. Good point - will change. 👍